### PR TITLE
Move settings array to Driver

### PR DESCRIPTION
**Pull request by omalizadeh**
Moving settings array to Base Driver and calling parent constructor from implemented drivers.

### CODE REVIEW

**Comment by tzsk** [comment]
This looks really nice.

It might be a better option to create an empty `boot()` method in abstract class Driver and calling that inside Driver Constructor.

Then other Driver Implementations can overwrite that boot method for initialisation instead of remembering to call the parent constructor.

We can update the docs and say it'll auto-set the settings array.

What do you think?

**Comment by omalizadeh** [comment]
yeah, that's much better. I make the changes.

File: src/Contracts/Driver.php [hunk 1]
```diff
@@ -42,5 +48,7 @@ public function message(string $message): Driver
         return $this;
     }
 
+    abstract protected function boot(): void;
```
Comment (tzsk) [inline src/Contracts/Driver.php:51]: Not every driver will need a boot method. It might be better to not make it `abstract`.

Comment (omalizadeh) [reply src/Contracts/Driver.php:51]: 🤔  yes. it's a possibility. Not forcing devs to implement it is a good option by the way.

**Comment by omalizadeh** [comment]
@tzsk

**Review by tzsk — APPROVED** [review]

**Pull request merged**